src/c++/vdo/base/dm-vdo-target.c

```diff
@@ -448,6 +451,43 @@ static inline int __must_check parse_bool(const char *bool_str, const char *true
 	return VDO_SUCCESS;
 }
 
+/**
+ * parse_compression() - Parse a conpression configuration string.
+ * @string: The string value describing compression options.
+ * @config: The configuration data structure to update.
+ *
+ * Return: VDO_SUCCESS or an error.
+ */
+static int __must_check parse_compression(const char *string,
+					  struct device_config *config)
+{
+	char *delimiter = strchr(string, ':');
+	char *option_string;
+	int key_length = strlen(string) - 1;
+	int result;
+
+	if (delimiter) {
+		key_length = delimiter - string;
+		option_string = &delimiter[1];
+	}
+
+	if ((strncmp(string, VDO_COMP_LZ4, key_length) == 0) && delimiter) {
+		result = kstrtoint(&delimiter[1], 10,
+				   &config->compression_level);
```

Comment:
if I'm reading the code right, "lz4:" will get parsed as compression level 0 -- kstrtoint seems to parse "\0" as 0, which is probably not what we want.